[Machine-learning][ACR-1748] Update ml-service documentation (#1809)
Samsung/tizen-docs: docs/application/native/guides/machine-learning/machine-learning-service.md

# Machine Learning Service

The Machine Learning Service API provides utility interfaces for both AI developers and application developers. AI developers can provide their pipelines and models as ML services, and application developers can use these services via this API.

The main features of the Machine Learning Service API include the following:

- Provide AI pipelines and models as ML services via machine-learning-agent (daemon).

  AI developers can manage their AI pipeline description with a unique name. Then they can launch the pipeline as an ML service. They also provide their model files with a unique name. Application developers can access and use these model files.

> - APIs for [Pipeline](#machine-learning-service-api-for-pipeline) and [Model](#machine-learning-service-api-for-model)

Reviewer: ```suggestion
- APIs for [pipeline](#machine-learning-service-api-for-pipeline) and [model](#machine-learning-service-api-for-model)
```

    ml_tensors_data_destroy (output);
    ```

## Machine Learning Service API for model

AI developers and application developers work together to manage an ML-powered application. AI developers focus on improving the performance or accuracy of model file. On the other hand, application developers do not care about details of the model file itself as long as the layout of model is preserved. Decoupling those two groups can improve development efficiency in terms of DevOps.

Let's say an application utilizes the famous MobileNet based model for its image classification task. While maintaining it, AI developers keep improving the accuracy or performance of the model. Instead of changing the application code to handle this model update, the application may use the latest registered MobileNet model. Machine Learning Service API supports this application development scenario.

> [!NOTE]

> > Machine Learning Service API for model is supported since Tizen 7.5

Reviewer: ```suggestion
> Machine Learning Service API for model is supported since Tizen 7.5.
```

    ml_tensors_data_destroy (output);
    ```

## Machine Learning Service API for Model

AI developers and application developers work together to manage an ML-powered application. AI developers focus on improving the performance or accuracy of model file. On the other hand, application developers do not care about details of the model file itself as long as the layout of model is preserved. Decoupling those two groups can improve development efficiency in terms of DevOps.

Let's say an application utilizes the famous mobilenet based model for its image classification task. While maintaining it, AI developers keep improving the accuracy or performance of the model. Instead of changing application code to handle this model update, application may use the latest registered mobilenet model. Machine Learning Service API supports this application development scenario.

> 

Reviewer: > [!NOTE]
> Machine Learning Service API for model is supported since Tizen 7.5

### AI developers registering their model

> AI developers can release a model providing application. The application has model file and register it using this API when the application is launched:

Reviewer: ```suggestion
AI developers can release a model providing application. The application has a model file, and it is registered using this API when the application launches:
```

guint info_num;
ml_service_model_get_all (key, &all_info_list, &info_num);

for (int i = 0; i < info_num; i++) {
  // Do something for each model info
  ...
}

// Delete the model from the ml service
ml_service_model_delete (key, version);
```

### Application code using the registered model

Application developers can use the registered model:

```c
// The name shared with AI developers
const gchar *key = "imgcls-mobilenet";
gchar *model_path;

ml_option_h activated_model_info;

// Get registered and activated model via ML Service API
int status = ml_service_model_get_activated (key, &activated_model_info);
if (status == ML_ERROR_NONE) {
  // If there is a model registered by AI developers, use it
  gchar *activated_model_path;

  // Get path of the model
  status = ml_option_get (activated_model_info, "path", (void **) &activated_model_path);

  model_path = g_strdup (activated_model_path);
} else {
  // If there is no registered model, use the default model in the app
  model_path = g_strdup_printf ("%s/%s", app_get_resource_path (), "mobilenet_v1.tflite");
}

// Do inference with the variable `model_path`
...

```

## Related information

- Dependencies

> 
>     Machine Learning Service API for Pipeline
>     - Tizen 7.0 and Higher for Mobile
>     - Tizen 7.0 and Higher for Wearable
>     - Tizen 7.0 and Higher for IoT

Reviewer: ```suggestion
    - Tizen 7.0 and Higher for Mobile
    - Tizen 7.0 and Higher for Wearable
    - Tizen 7.0 and Higher for IoT
```

> 
>     Machine Learning Service API for Model
>     - Tizen 7.5 and Higher for Mobile
>     - Tizen 7.5 and Higher for Wearable
>     - Tizen 7.5 and Higher for IoT

Reviewer: For dependencies, it is recommended to mention the lowest of the supported version.
For model, it would be better to use an alert to mention the version of support.
```suggestion
```
